Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.Contracts.Core.AttachAnnexToBindingContract.BusinessRules;
+
+using EvolutionaryArchitecture.Fitnet.Common.Core.BusinessRules;
+
+internal sealed class AnnexCanOnlyBeAttachedToActiveBindingContractRule : IBusinessRule
+{
+    private readonly DateTimeOffset? _bindingContractTerminatedAt;
+    private readonly DateTimeOffset _bindingContractExpiringAt;
+    private readonly DateTimeOffset _annexValidFrom;
+
+    internal AnnexCanOnlyBeAttachedToActiveBindingContractRule(
+        DateTimeOffset? bindingContractTerminatedAt,
+        DateTimeOffset bindingContractExpiringAt,
+        DateTimeOffset annexValidFrom)
+    {
+        _bindingContractTerminatedAt = bindingContractTerminatedAt;
+        _bindingContractExpiringAt = bindingContractExpiringAt;
+        _annexValidFrom = annexValidFrom;
+    }
+
+    public bool IsMet() => !_bindingContractTerminatedAt.HasValue && _bindingContractExpiringAt > _annexValidFrom;
```

Comment:
For me this logic match to rule

!_bindingContractTerminatedAt.HasValue && _bindingContractExpiringAt.


This logic is different rule: Annex Validation Date Should Be In Contract LifeTime (or something like that)
_bindingContractExpiringAt > _annexValidFrom 

What you think?

Reply:
Done in a bit different way

Reply:
@kamilbaczek any feedback on new version?

Reply:
Looks very well for me! Thanks for this change